docs/concepts/registries.md

```diff
@@ -0,0 +1,82 @@
+Registries
+==========
+
+Registration is the process of taking the objects of a mod (items, blocks, sounds, etc.) and making them known to the game. Registering things is important, as without registration the game will simply not know about these objects in a mod and will exhibit great amounts of unexplainable (and probably crashy) behavior.
+
+Most things that require registration in the game are handled by the Forge registries. A registry is a simple object similar to a map that assigns values to keys. Forge uses registries from `ResourceLocation`s to registry objects, which are of type `IForgeRegistryEntry`. This allows the RL to act like a "registry name" for the object. The registry name for an object may be accessed with `get`/`setRegistryName`. The setter can only ever be called once, and calling it twice results in an exception. Every type of registrable object has its own registry, and names in two different registries will not collide. (E.g. there's a registry for `Block`s, and a registry for `Item`s, and a `Block` may be registered with the name `mod:example`, and an `Item` may be registered with the same name, without colliding. However, if two blocks were registered with that name, the game would crash.) The default implementation of `IForgeRegistryEntry` also provides two convenience implementations of `setRegistryName`: one where the parameter is a single string, and one where there are two string parameters. The overload that takes a single string checks whether the input contains a `:` (i.e. it checks whether the passed in stringified `ResourceLocation` has a domain), and if it doesn't it uses the current modid as the resource domain. The two argument overload simply constructs the registry name using the `modID` as the domain and `name` as the path.
```

Comment:
Throughout the docs "modid" is used, so I'll leave it like that. I searched for all usages of all forms of that word with `grep -E '[Mm]od\s?[Ii][Dd]' -r docs/` and found it the most common.